packages/core/src/core.ts

```diff
@@ -1216,6 +1216,8 @@ class ThresholdKey implements ITKey {
     } else {
       throw CoreError.default("can only add type ShareStore into shares");
     }
+    ss = this.metadata.shareToShareStore(ss.share.share as BN);
```

Comment:
would we have simpler check if we check for existence of the input shareStore's polyID in the polyIDList ?
@metallicalfa2  

also, should we add check for the `inputShareStore` that the polyID should be the latest polyID?

Reply:
I think the shareToShareStore won't fix the issue.
we should check the if latest share's metadata polynomial has the current tkey latest polynomial.

Reply:
negative test cases are working fine, can you explain which case can fail 
https://github.com/tkey/tkey/pull/210/files#diff-d264ebdc71dfea141ff0afde527572308cf1712526e641e7598b4e56ca3e4780R896-R899

Reply:
Possible scenario,

create 2 tkey instance, InstA, InstB
InstB generate new share E, 
input share E to tkey InstA.
shareToShareStore would failed as share E is not expected in InstA metadata